### PR TITLE
Add custom Datepicker styles and validation

### PR DESCRIPTION
**Pull request by psealock**
**Designs** p6riRB-3hq-p2

This PR add the styles for the custom datepicker.

## Also Added
* A reset button for the custom datepicker
* Date input validation, including a tooltip error message
* Tablet size screens will render two calendars

## Screenshots
screen shot 2018-07-25 at 4 54 44 pm
...
screen shot 2018-07-25 at 4 54 58 pm
...
screen shot 2018-07-25 at 4 55 38 pm
...
screen shot 2018-07-25 at 4 56 15 pm
...
screen shot 2018-07-25 at 4 56 46 pm
...
screen shot 2018-07-25 at 4 57 09 pm

@LevinMedia could you take this for a spin? I departed from your designs in the error border on the invalid text input in favor of inset box shadow. This is so focus styles can still be seen. Notice I only show the tooltip when the input has focus.

## Test

1. `/wp-admin/admin.php?page=wc-admin#/analytics/products`
2. Poke around the datepicker functionality including validation, keyboard input, reseting, and mobile styles
3. Note: the error popover is placed incorrectly when not viewing in `is-mobile`. Gutenberg incorrectly calculates position of a popover inside of another popover. The cause seems to be [a parent element using CSS `transform` which creates a new context making subsequent child `position: fixed;` elements relative to the parent, not the viewport](https://coderwall.com/p/ka0low/don-t-mix-position-fixed-with-css-transforms). I will look into this separately.
screen shot 2018-07-25 at 5 06 14 pm
4. Lastly, primary button styles to be handles separately in https://github.com/woocommerce/wc-admin/issues/145

### CODE REVIEW

File: client/components/date-picker/content.js [hunk 1]
```diff
@@ -89,26 +93,55 @@ class DatePickerContent extends Component {
...
+												isPrimary
+												onClick={ resetCustomValues }
+												disabled={ ! ( after || before ) }
+											>
+												{ __( 'Reset', 'wc-admin' ) }
+											</Button>
+										) }
+										{ isValidSelection( selectedTab ) ? (
+											<Link
+												className={ classnames(
```
Comment (ryelle) [inline client/components/date-picker/content.js]: If all the classNames are static, you can just pass them in as a string rather than using `classNames()`

**Comment by ryelle** [comment]
I also had a problem with seeing the buttons at first - my screen was too short, and the buttons are cut off without scroll:

screen shot 2018-07-25 at 3 20 19 pm

**Comment by psealock** [comment]
Thanks @ryelle 

* The popover content is now scrollable on short screens
* Improved contrast on inputs. `$core-grey-dark-500` on text and `$core-grey-dark-300` on placeholders.
* Removed unneeded use of `classnames`

File: client/components/date-picker/content.js [hunk 2]
```diff
@@ -16,6 +17,8 @@ import PresetPeriods from './preset-periods';
 import Link from 'components/link';
 import { DateRange } from 'components/calendar';
 
+const isMobileViewport = () => window.innerWidth < 782;
```
Comment (timmyc) [inline client/components/date-picker/content.js]: Is there already an event somewhere that will trigger a re-render on change of window size - or do we need to add logic here to handle that

**Comment by ryelle** [comment]
Height is good on the calendar now, it scrolls on my shorter browser :) The contrast on the inputs is also 👍 , but I don't think the contrast on the day-of-week headers, `Mo Tu We` etc, has been updated, it's still just a hair below passing contrast.

File: client/components/date-picker/content.js (hunk 2 above)
Comment (psealock) [reply client/components/date-picker/content.js]: There isn't a listener to trigger re-render on resize, but this calculation is done on every opening of the custom tab as you noted. My opinion is the use of another event listener doesn't justify the benefit of being able to render two calendars when the user is resizing with the custom date tab open. How often will this happen? What do you think @timmyc ?

**Comment by psealock** [comment]
>but I don't think the contrast on the day-of-week headers, Mo Tu We etc, has been updated

Oops, missed that. Updated now to ` $core-grey-dark-400` as suggested

File: client/components/date-picker/content.js (hunk 2 above)
Comment (timmyc) [reply client/components/date-picker/content.js]: I think we are fine with it as-is. Very much an edge case, and it works fine just the way it is when  you resize the browser.

**Comment by timmyc** [comment]
Marking this as ready to merge, seems like 839eb2c addressed @ryelle's last request... but please chime in if there are other items.

**Comment by ryelle** [comment]
Yep, it's good now! (didn't see the comment notification 😣 )

**Pull request merged**